SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift

```diff
@@ -599,14 +600,14 @@ class SourceKitDocument {
     switch request {
     case .editorReplaceText(_, let offset, let length, let text):
       let edit = IncrementalEdit(range: ByteSourceRange(offset: offset, length: length), replacementLength: text.utf8.count)
-      reparseTransition = IncrementalParseTransition(previousTree: self.tree!, edits: ConcurrentEdits(edit))
+      reparseTransition = IncrementalParseTransition(previousTree: self.tree!, edits: ConcurrentEdits(edit), lookaheadRanges: self.lookaheadRanges!)
     default:
       reparseTransition = nil
     }
 
     var tree: SourceFileSyntax
     if let state = sourceState {
-      tree = Parser.parse(
+      (tree, self.lookaheadRanges) = Parser.parseIncrementally(
```

Comment:
Real nitpick: We should be consistently using or not using `self.` here. Can you either drop it from `self.lookaheadRanges` or add it to `tree`?

Reply:
Sure.